tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -177,59 +185,50 @@ public int PartialFlushMinSpans
         /// </summary>
         internal MetricsTransportType MetricsTransport { get; set; }
 
-        private void ConfigureMetricsTransport(IConfigurationSource source, bool forceMetricsOverUdp)
+        private void ConfigureMetricsTransport(IConfigurationSource source)
         {
-            MetricsTransportType? metricsTransport = null;
-
+            var agentHost = source?.GetString(ConfigurationKeys.AgentHost);
             var dogStatsdPort = source?.GetInt32(ConfigurationKeys.DogStatsdPort);
+            var metricsPipeName = source?.GetString(ConfigurationKeys.MetricsPipeName);
+            var metricsUnixDomainSocketPath = source?.GetString(ConfigurationKeys.MetricsUnixDomainSocketPath);
 
-            MetricsPipeName = source?.GetString(ConfigurationKeys.MetricsPipeName);
-
-            // Agent port is set to zero in places like AAS where it's needed to prevent port conflict.
-            // The agent will fail to start if it can not bind a port.
-            // If the dogstatsd port isn't explicitly configured, check for pipes or sockets.
-            if (!forceMetricsOverUdp && (dogStatsdPort == 0 || dogStatsdPort == null))
+            if ((dogStatsdPort != null && dogStatsdPort != 0) || agentHost != null)
```

Comment:
If we're ignoring `0`, do we want to ignore negative numbers as well`?
(and out of scope here, but should we log invalid settings?)
```suggestion
            if ((dogStatsdPort != null && dogStatsdPort > 0) || agentHost != null)
```

Reply:
In the current state of this PR, negative ports will be logged indirectly as we will log a warning when trying to create the URL.
But you're right maybe I shall start by checking all variables, see if they are valid, log and ignore them if not valid 🤔